dialog/file.go

```diff
@@ -555,15 +554,22 @@ func (f *fileDialog) setView(view ViewLayout) {
 			f.setSelected(file, id)
 		}
 	}
+
+	var toggleViewButtonIcon fyne.Resource
 	if f.view == GridView {
 		grid := widget.NewGridWrap(count, template, update)
 		grid.OnSelected = choose
 		f.files = grid
+		toggleViewButtonIcon = theme.ListIcon()
 	} else {
 		list := widget.NewList(count, template, update)
 		list.OnSelected = choose
 		f.files = list
+		toggleViewButtonIcon = theme.GridIcon()
 	}
+
+	f.toggleViewButton.SetIcon(toggleViewButtonIcon)
```

Comment:
I think it's 2 lines less code if you just call `SetIcon` instead of adding the `toggleViewButtonIcon` variable in this context

Reply:
Thanks. I also found another occurrence that seemed too verbose